ocs_ci/ocs/resources/pod.py

```diff
@@ -2547,24 +2547,51 @@ def check_toleration_on_pods(toleration_key=constants.TOLERATION_KEY):
 
     """
 
-    pod_objs = get_all_pods(
+    pod_objs = get_running_state_pods(
         namespace=config.ENV_DATA["cluster_namespace"],
-        selector=[constants.TOOL_APP_LABEL],
-        exclude_selector=True,
     )
-    flag = False
+
     for pod_obj in pod_objs:
         resource_name = pod_obj.name
         tolerations = pod_obj.get().get("spec").get("tolerations")
-        for key in tolerations:
-            if key["key"] == toleration_key:
-                flag = True
-        if flag:
-            logger.info(f"The Toleration {toleration_key} exists on {resource_name}")
-        else:
-            logger.error(
-                f"The pod {resource_name} does not have toleration {toleration_key}"
-            )
+        # Check if the specified toleration key is present in the tolerations
+        toleration_found = any(
+            toleration.get("key") == toleration_key for toleration in tolerations
+        )
+        # Check if the toleration key is not found
+        assert (
+            toleration_found
+        ), f"The pod {resource_name} does not have toleration {toleration_key}"
+        logger.info(f"The Toleration {toleration_key} exists on {resource_name}")
+
+
+def check_toleration_on_subscriptions(toleration_key=constants.TOLERATION_KEY):
+    """
+    Function to check toleration on subscriptions
+
+    Args:
+        toleration_key (str): The toleration key to check
+
+    """
+
+    sub_list = ocp.get_all_resource_names_of_a_kind(kind=constants.SUBSCRIPTION)
```

Comment:
you can pass the namespace as namespace=config.ENV_DATA["cluster_namespace"] while creating an instance. otherwise it will pull only the sub from the current namespace